### ISSUE TITLE
Music production suite

### ISSUE THREAD

**Issue by binarica** [helloSystem/hello#139]
Consider adding the following (or similar) applications to the base system to provide a basic, out-of-the-box, QT-based music production toolkit (idea borrowed from [Slackermedia Workflows](http://slackermedia.info/workflows/)):

- Digital Audio Workstation: Ardour/Rosegarden/LMMS (already listed on [Libre Multimedia Suite](https://github.com/AppImage/AppImageKit/wiki/Libre-Graphics-Suite))
- Drum Machine: Hydrogen
- Plugin Host: Carla
- Audio Synthesis/Live Coding: SuperCollider
- Scorewriter/Tab Editor: MuseScore
- JACK Frontend: QjackCtl

These are all available as FreeBSD ports without Linux emulation.

**Comment by probonopd** [comment]
hello @binarica.
Thanks for suggesting a curated set of music applications.

> Ardour/Rosegarden/LMMS

If you had to pick one, which one would it be? We are currently thin-installing LMMS. Or are they all usually used by the same user in conjunction (like Inscape for vector graphics and GIMP for pixel graphics)?

> QjackCtl

This is a controversial one. If I had to pick one piece of software that is probably at the exact other end of the "it just works without configuration" spectrum, I'd probably pick JACK. It never works for me right out of the box without having to fiddle around with it, and I find it less than intuitive. Hence, helloSystem does not use JACK by default (users who are advanced enough to understand JACK should have no problem installing and configuring it). Or am I missing something? I don't need much - I would already be happy if sound input and output would each automatically switch to the device last plugged in, with the ability to change that selection system-wide.

**Comment by ILoveGadgets** [comment]
Hi @probonopd 
The problem with Ardour/Rosegarden/LMMS is that while they have some things in common they each specialise in a different area.

I have used LMMS to make drum/bass tracks to play along to with guitar, but if I wanted to record that guitar I would pick Ardour as it is better suited for the task.

They are very much like the different graphics, DTP and CAD programs, you would use more than one at different stages of a project.

I would pick LMMS and Ardour, as they have modern interfaces that go well together, Rosegarden looks a bit dated and at least to me looks cluttered and "noisy" on screen.

**Comment by probonopd** [comment]
Well...

```
Users-RevoOne-RL85% sudo pkg install ardour5
Updating FreeBSD repository catalogue...
FreeBSD repository is up to date.
All repositories are up to date.
pkg.real: No packages available to install matching 'ardour5' have been found in the repositories
```

**Comment by probonopd** [comment]
Looks like Ardour wants JACK, and every time something needs JACK things fail badly for me. JACK always results in a never ending odyssee of fiddling around in configurations and errors upon errors. So no Ardour for me - it just doesn't work right out of the box:

![image](https://user-images.githubusercontent.com/2480569/109062065-bae49e80-76e7-11eb-814d-2c1a4bb6f83a.png)

**Comment by 69graves** [comment]
I concur with the above post.  My problems with Jack and my inability to be able to just sit down at a Linux computer to record music pushed me eventually in the direction of buying a Mac and Logic Pro X.

Were I to pick a good place to start, it would be Audacity and perhaps a revival of the once promising looking [Jokosher.](https://www.jokosher.org)

I don't know the complexities well enough to be able to assess if it would be feasible to do so, but there has to be someone out there that can figure out.

All this said, with the goal of helloSystem to be ease of use, I would also vote against implementing Jack.

**Comment by probonopd** [comment]
Yes, I am using Audacity (without JACK) on helloSystem.

JACK is probably "professional grade" and possibly even more so than audio on the Mac (although I don't know this for sure); possibly all that is missing is a really user-friendly interface. Maybe someone "just" needs to write it.

**Comment by acook** [comment]
I just wanted to mention to those who found Linux unsuitable for music production that the commercial program BitWig works seamlessly for me. Unfortunately I don't think they intend to build a BSD version of it.

More on topic, I think Audacity and LMMS is probably adequate for now. At least until someone is willing to tackle the JACK problem.

**Comment by Lestibournes** [comment]
Is PipeWire something you'd be looking into using?

**Comment by Lestibournes** [comment]
Jokosher looked fantastic, but it was s GTK2 app based on GStreamer IIRC.

**Comment by probonopd** [comment]
> Is PipeWire something you'd be looking into using?

No, https://github.com/helloSystem/hello/wiki/Welcome-and-unwelcome-technologies

**Comment by probonopd** [comment]
**Virtual OSS** has been mentioned as something to look into.

**Comment by plastmassor33** [comment]
> > Is PipeWire something you'd be looking into using?
> 
> No, https://github.com/helloSystem/hello/wiki/Welcome-and-unwelcome-technologies

No experience with FreeBSD, but I have to share my experience as a linux musician:
Before Pipewire only simple music programs, like Musescore (which uses only soundfonts) and the common apps like the web browser, music player, etc. worked as expected. 
With Pipewire **everything** works as expected (and it has nothing to do with Wayland).

Before Pipewire Audacity would start randomly glitching and crashing after a period of using it (I remember seeing underruns and overruns when launched through terminal). Anything live audio (e.g. a VST effect) would always be crackling. JACK indeed is possibly one of the hardest things to setup in the universe and when I finally managed to get it running it actually solved NONE of the problems. All this made me actively question if it's even possible to work with audio on Linux. 

But with Pipewire now enabled by default on Ubuntu everything just works as expected. It's so refreshing for once not to have to think about which API to use and let each program use it's default, yet still have access to the power of JACK's midi/audio routing through something like qpwgraph when needed.
